Adapt UI to new json-rpc functions

edisontim opened the pull request:
- Gets discussions when getting into a Realm for the first time after logging in
- Fetches NPCs backstories
- Changed `townhall` to `discussions`, left some explicitely as town hall fit more for the user sometimes

cwastche commented on client/src/components/cityview/realm/villagers/panels/discussion/NpcChatMessage.tsx:
```diff
@@ -86,7 +83,7 @@ export const NpcChatMessage = (props: NpcChatMessageProps) => {
         </div>
       </div>
-      {showNpcStats && <NpcPopup onClose={() => setShowNpcStats(false)} npc={npc} />}
+      {showNpcStats && <NpcPopup onClose={() => setShowNpcStats(false)} />}
```
This would create a popup for each villager in the discussion, thus making the user able to open a lot of them. What do you think about setting the NpcPopup higher in the div elements hierarchy to replace the content of the popup instead of opening a new one, much like the implementation in the villagers panel ?

cwastche reviewed (commented):
The changes make it way more readable, nice one !
There's a few ways we could manage the NpcPopups, but I think making them have the same behaviour between both panels is necessary.

edisontim commented:
- Added a commit to remove some misplaced elements from the `NpcStore`. Added a `DiscussionContext` instead. 
- Removed `selectedNpc` and function to set the state from `NpcStore` and use them as props now
